doc/HowToPipeline.md

```diff
@@ -0,0 +1,2040 @@
+        self.mapper.addMinimalSchema(inputSchema, True)
+
+        # Add our new field
+        self.apKey = self.mapper.editOutputSchema().addField("apFlux", type=np.float64,
+                                                             doc="Ap flux measured")
+
+        # Get the output schema
+        self.schema = mapper.getOutputSchema()
+    
+        # Create the output catalog
+        self.outputCatalog = afwTable.SourceCatalog(self.schema)
+
+        # Put the outputSchema into a SourceCatalog container. This var name matches an initOut
+        # so will be persisted
+        self.outputSchema = afwTable.SourceCatalog(self.schema)
+
+    def run(self, exposure: afwImage.Exposure, inputCatalog: afwTable.SourceCatalog,
+            background: typing.Union[None, afwMath.BackgroundList] = None) -> pipeBase.Struct:
+        # If a background is supplied, add it back to the image so local background subtraction
+        # can be done.
+        if background is not none:
+            exposure.image.array += background.getImage()
+
+        # Add in all the records from the input catalog into what will be the output catalog
+        self.outputCatalog.extend(inputCatalog, mapper=self.mapper)
+
+        # Get the dimensions of the exposure
+        dimensions = exposure.getDimensions()
+
+        # Get indexes for each pixel
+        indy, indx = np.indices((dimensions.getY(), dimensions.getX()))
+
+        # Loop over each record in the catalog
+        for source in outputCatalog:
+            # Create an aperture and measure the flux
+            center = source.getCentroid()
+            mask = ((indy - center.getY())**2 + (indx - center.getX())**2)**0.5 < self.apRad
+            flux = np.sum(exposure.image.array*mask)
+
+            # Do local background subtraction
+            if background is not None:
+                outerAn = distance < 2.5*self.apRad
+                innerAn = distance < 1.5*self.apRad
+                annulus = outerAn - innerAn
+                localBackground = np.mean(exposure.image.array*annulus)
+                flux -= np.sum(mask)*localBackground
+
+            # Set the flux field of this source
+            source.set(self.apKey, flux)
+        
+        return pipeBase.Struct(outputCatalog=self.outputCatalogs)
+```
+
+## Prerequisite inputs
+Some tasks make use of data-sets that are created outside the processing environment of LSST data-sets.
+These may be things like reference catalogs, bright star masks, calibrations, etc. To account for this
+PipelineTasks have a special type of connection called, `PrerequisiteInput`. This type of input tells the
+execution system, that this is a special type of data-set, and not to try and expect it to be produced
+anywhere in a processing pipeline. If this data-set is not found, the system will raise a hard error and
+tell you the data-set type is missing instead of inferring that there is some processing step that is
+missing.
```

Comment:
"inferring that some processing step that is missing" -> "inferring that some other processing step will produce these datasets"